include/deal.II/base/bounding_box_data_out.h

```diff
@@ -0,0 +1,159 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_bounding_box_data_out_h
+#define dealii_bounding_box_data_out_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/data_out_base.h>
+#include <deal.II/base/geometry_info.h>
+#include <deal.II/base/iterator_range.h>
+#include <deal.II/base/point.h>
+
+#include <deal.II/boost_adaptors/bounding_box.h>
+#include <deal.II/boost_adaptors/point.h>
+#include <deal.II/boost_adaptors/segment.h>
+
+DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
+#include <boost/geometry/index/rtree.hpp>
+#include <boost/geometry/strategies/strategies.hpp>
+DEAL_II_ENABLE_EXTRA_DIAGNOSTICS
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * This class generates graphical output for BoundingBox objects, starting from
+ * any object that can be converted by boost to a BoundingBox.
+ *
+ * @author Luca Heltai, 2020.
+ */
+template <int dim>
+class BoundingBoxDataOut : public DataOutInterface<dim, dim>
+{
+public:
+  BoundingBoxDataOut() = default;
+
+  ~BoundingBoxDataOut() = default;
+
+  /**
+   * Generate patches from a range of objects that can be converted by boost to
+   * a collection of BoundingBox objects.
+   */
+  template <class ConvertibleToBoundingBoxIterator>
+  void
+  build_patches(const ConvertibleToBoundingBoxIterator &begin,
```

Comment:
What are examples of these function arguments? Just iterators over collections of `BoundingBox` type, or are there other things that are convertible to `BoundingBox` objects?